File: app/src/main/java/com/studyasist/data/local/entity/AttemptAnswer.kt
package com.studyasist.data.local.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "attempt_answers",
    foreignKeys = [
        ForeignKey(
            entity = Attempt::class,
            parentColumns = ["id"],
            childColumns = ["attemptId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index(value = ["attemptId"])]
)
data class AttemptAnswer(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val attemptId: Long,
    val qaId: Long,
    val answerText: String? = null,
    val answerImageUri: String? = null,
    val answerVoiceUri: String? = null,
    val submittedAt: Long
)

File: StudyAsist-iOS/StudyAsist/Models/AttemptAnswer.swift
//
//  AttemptAnswer.swift
//  StudyAsist
//

import Foundation
import SwiftData

@Model
final class AttemptAnswer {
    @Attribute(.unique) var id: Int64
    var attemptId: Int64
    var qaId: Int64
    var answerText: String?
    var answerImageUri: String?
    var answerVoiceUri: String?
    var submittedAt: Int64
    var timeSpentSeconds: Int
    
    init(
        id: Int64 = 0,
        attemptId: Int64,
        qaId: Int64,
        answerText: String? = nil,
        answerImageUri: String? = nil,
        answerVoiceUri: String? = nil,
        submittedAt: Int64,
        timeSpentSeconds: Int = 0
    ) {
        self.id = id
        self.attemptId = attemptId
        self.qaId = qaId
        self.answerText = answerText
        self.answerImageUri = answerImageUri
        self.answerVoiceUri = answerVoiceUri
        self.submittedAt = submittedAt
        self.timeSpentSeconds = timeSpentSeconds
    }
}
